### ISSUE TITLE
"none" feature not dropped when merging inferred accessibility features

### ISSUE THREAD

**Issue by sylvain-degeilh-lingsoft** [readium/cli#102]
Hi,

We’re using the RWP manifest CLI to extract accessibility metadata from EPUBs, with the "merged" option turned on so missing properties get inferred.

It looks like something might be off with how the merging works. In one example, the EPUB explicitly sets "feature": "none", but it clearly has a table of contents. The CLI ends up returning: 
"feature": ["none", "tableOfContents"]

We would expect:
"feature": ["tableOfContents"]

So it seems like "none" isn’t being removed when other features are inferred, which leads to a confusing result.

Other options ("no", "split") return "feature": ["none"].

Tested on a clean install 
- v0.6.6 of the readium cli (go-toolkit v0.13.4)
- go version 1.24.2
- Ubuntu 24.04 LTS
With the attached file moby_dick_af_none.epub

Thanks,
Sylvain

PS: The attached file moby_dick_af_none.zip is in fact an epub file. The extension has been changed to zip as .epub is not accepted.

[moby_dick_af_none.zip](https://github.com/user-attachments/files/26245955/moby_dick_af_none.zip)

**Comment by HadrienGardeur** [comment]
Thanks for this report.

When metadata are merged, I think that we should indeed drop `none` but when split, having `none` in extracted metadata and something else in inferred metadata is more of a feature than a bug IMO.

In general, I would recommend using the split mode though, this way you can differentiate in your system what was extracted vs inferred.
